### PR TITLE
Throw error on duplicate spa-module-attr assignment

### LINKED ISSUE #897: Re-addding a component to spa.SPA chrashes it
Minimal example from Jupyter notebook:
```python
D=16
vocab1 = spa.Vocabulary(D)
vocab1.parse('A+B+C')

with spa.SPA('AssociativeMemory', seed=1) as model_1:
    model_1.assoc_mem = spa.AssociativeMemory(input_vocab=vocab1)
```

In another cell:
```python 

with model_1:
    model_1.net_input = spa.Input(assoc_mem='A')
```

When **re-running** the second cell the network crashes with:
```python
ValueError: <Input "net_input"> was not added as an attribute of the SPA network and won't be detected
```

### PR DESCRIPTION
**Pull request by Seanny123**
Resolves #897

### CODE REVIEW

**Comment by jgosmann** [comment]
Didn't we say that the error should only be raised if the attribute already is a module? (e.g. if the attribute is an integer allow it to be changed)

**Comment by Seanny123** [comment]
Oops. Will fix

**Comment by Seanny123** [comment]
Fixed now

**Comment by Seanny123** [comment]
@jgosmann do you want to take another look at this now that I fixed the error?

**Comment by jgosmann** [comment]
LGTM

File: nengo/spa/tests/test_spa.py [hunk 1]
```diff
@@ -9,12 +9,19 @@ def test_spa_verification(Simulator, seed, plt):
...
 
     model = spa.SPA(seed=seed)
+
     # build a normal model that shouldn't raise a warning
     with model:
         model.buf = spa.Buffer(d)
         model.input_node = spa.Input(buf='B')
         # make sure errors aren't fired for non-spa modules
         prod = nengo.networks.Product(10, 2)  # noqa: F841
+        model.int_val = 1
```
Comment (tbekolay) [inline nengo/spa/tests/test_spa.py]: What's this doing here?

File: nengo/spa/spa.py [hunk 2]
```diff
@@ -98,6 +98,10 @@ def __setattr__(self, key, value):
         This is so that we can use the variable name for the Module as
         the name that all of the SPA system will use to access that module.
         """
+        if hasattr(self, key) and isinstance(getattr(self, key), Module):
+            raise ValueError("Cannot re-assign module-attribute %s to %s. "
+                             "SPA module-attributes can only be assigned"
```
Comment (tbekolay) [inline nengo/spa/spa.py]: Need a space at the end of this second line.

File: nengo/spa/tests/test_spa.py (hunk 1 above)
Comment (jgosmann) [inline nengo/spa/tests/test_spa.py]: It is testing that attribute assignment with non-spa module stuff still works without a warning?

Comment (tbekolay) [inline nengo/spa/tests/test_spa.py]: Ah, gotcha. That should happen twice right, to make sure it can be overwritten (unlike modules)?

**Comment by Seanny123** [comment]
fixed @tbekolay comments (:

**Pull request merged**